fiftyone/operators/builtin.py

```diff
@@ -434,90 +1835,428 @@ class LoadWorkspace(foo.Operator):
+            label="Load workspace",
+            dynamic=True,
         )
 
     def resolve_input(self, ctx):
         inputs = types.Object()
-        inputs.str("name", label="Workspace Name", required=True)
-        return types.Property(inputs)
+
+        workspaces = ctx.dataset.list_workspaces()
+
+        if workspaces:
+            workspace_selector = types.AutocompleteView()
+            for key in workspaces:
+                workspace_selector.add_choice(key, label=key)
+
+            inputs.enum(
+                "name",
+                workspace_selector.values(),
+                default=None,
+                required=True,
+                label="Workspace",
+                description="The workspace to load",
+                view=workspace_selector,
+            )
+        else:
+            prop = inputs.str(
+                "msg",
+                label="This dataset has no saved workspaces",
+                view=types.Warning(),
+            )
+            prop.invalid = True
+
+        return types.Property(inputs, view=types.View(label="Load workspace"))
 
     def execute(self, ctx):
-        name = ctx.params.get("name", None)
+        name = ctx.params["name"]
+
         ctx.ops.set_spaces(name=name)
-        return {}
 
 
 class SaveWorkspace(foo.Operator):
     @property
     def config(self):
         return foo.OperatorConfig(
-            name="save_workspace", label="Save Workspace"
+            name="save_workspace",
+            label="Save workspace",
+            dynamic=True,
         )
 
     def resolve_input(self, ctx):
         inputs = types.Object()
-        inputs.str("name", label="Workspace Name", required=True)
-        inputs.str("description", label="Description")
-        inputs.str("color", label="Color")
-        inputs.obj("spaces", label="Spaces")
-        inputs.bool("edit", label="Edit")
-        if ctx.params.get("edit", False):
-            inputs.str(
-                "current_name", label="Current Workspace Name", required=True
+
+        workspaces = ctx.dataset.list_workspaces()
+        workspace_selector = types.AutocompleteView()
+        for key in workspaces:
+            workspace_selector.add_choice(key, label=key)
+
+        inputs.str(
+            "name",
+            required=True,
+            label="Name",
+            description="A name for the saved workspace",
+            view=workspace_selector,
+        )
+
+        inputs.str(
+            "description",
+            default=None,
+            required=False,
+            label="Description",
+            description="An optional description for the workspace",
+        )
+
+        inputs.str(
+            "color",
+            default=None,
+            required=False,
+            label="Color",
+            description=(
+                "An optional RGB color string like `#FF6D04` for the workspace"
+            ),
+        )
+
+        # @todo infer this automatically from current App spaces
```

Comment:
@imanjra @ritch can do we have a way to get the current `session.spaces` layout during operator execution?

Reply:
No, not at the moment. It should be fairly trivial to add it in.

Reply:
If this could also include the opened panel names that would be super useful